examples/tabs.rs

```diff
@@ -80,30 +121,44 @@ fn run_app<B: Backend>(terminal: &mut Terminal<B>, mut app: App) -> io::Result<(
 
 fn ui(f: &mut Frame, app: &App) {
     let area = f.size();
-    let vertical = Layout::vertical([Constraint::Length(3), Constraint::Min(0)]);
+    let vertical = Layout::vertical([Constraint::Length(4), Constraint::Min(3)]);
     let [tabs_area, inner_area] = area.split(&vertical);
 
-    let block = Block::default().on_white().black();
-    f.render_widget(block, area);
-    let tabs = app
-        .titles
-        .iter()
-        .map(|t| {
-            let (first, rest) = t.split_at(1);
-            Line::from(vec![first.yellow(), rest.green()])
-        })
-        .collect::<Tabs>()
-        .block(Block::default().borders(Borders::ALL).title("Tabs"))
-        .select(app.index)
-        .style(Style::default().cyan().on_gray())
-        .highlight_style(Style::default().bold().on_black());
-    f.render_widget(tabs, tabs_area);
-    let inner = match app.index {
-        0 => Block::default().title("Inner 0").borders(Borders::ALL),
-        1 => Block::default().title("Inner 1").borders(Borders::ALL),
-        2 => Block::default().title("Inner 2").borders(Borders::ALL),
-        3 => Block::default().title("Inner 3").borders(Borders::ALL),
-        _ => unreachable!(),
-    };
-    f.render_widget(inner, inner_area);
+    render_tabs(f, app, tabs_area);
+
+    render_inner(f, app, inner_area);
+}
+
+fn render_tabs(f: &mut Frame, app: &App, area: Rect) {
+    let block = Block::new()
+        .title("Tabs Example".bold())
+        .title("Use h l or ◄ ► to change tab")
+        .title_alignment(Alignment::Center)
+        .padding(Padding::top(1)); // padding to separate tabs from block title.
+
+    let selected_tab_index = app.selected_tab as usize;
+    let titles = SelectedTab::iter().map(SelectedTab::tab_title);
+    let tabs = Tabs::new(titles)
```

Comment:
With the earlier impl you could very probably just give `Tabs::new(SelectedTab::iter())` or something similar.